PR for SRVLOGIC-530: Add the section to upgrade Serverless Logic Operator from 1.35.0 to 1.36.0 (#93332)
openshift/openshift-docs: serverless-logic/serverless-logic-managing-upgrades/serverless-logic-upgrading-operator-from-1.35-to-1.36.adoc

:_mod-docs-content-type: ASSEMBLY
[id="serverless-logic-upgrading-operator-from-1.35-to-1.36"]
= Upgrading OpenShift Serverless Logic Operator from version 1.35.0 to 1.36.0
include::_attributes/common-attributes.adoc[]
:context: serverless-logic-upgrading-operator-from-1.35-to-1.36

toc::[]

> This section provides step-by-step instructions to upgrade the {ServerlessLogicOperatorName} from version 1.35.0 to 1.36.0. The upgrade process involves preparing the existing workflows and services, updating the Operator, and restoring the workflows after the upgrade.

Reviewer: ```suggestion
You can upgrade the {ServerlessLogicOperatorName} from version 1.35.0 to 1.36.0. The upgrade process involves preparing the existing workflows and services, updating the Operator, and restoring the workflows after the upgrade.
```

> Avoid self-referential language, such as "This topic covers…​" or "Use this procedure to…​".
- https://redhat-documentation.github.io/supplementary-style-guide/#shortdesc

[NOTE]
====

> Different workflow profiles require different upgrade steps. Carefully follow the instructions for each profile.

Reviewer: ```suggestion
Different workflow profiles require different upgrade steps. Follow the instructions for each profile carefully.
```

The preparation process includes:

* Deleting or scaling workflows based on their profiles.
* Backing up all necessary databases and resources.
* Ensuring you have a record of all custom configurations.

.Prerequisites

* You have access to the cluster with `cluster-admin` privileges.
* You have {ServerlessLogicOperatorName} installed on your cluster.
* You have access to a {ServerlessLogicProductName} project with the appropriate roles and permissions to create applications and other workloads in {ocp-product-title}.
* You have access to the *OpenShift Management Console* for Operator upgrades.
* You have installed the OpenShift CLI (`oc`).

include::modules/serverless-logic-upgrade-1.36-deleting-workflows-with-dev-profile.adoc[leveloffset=+2]

include::modules/serverless-logic-upgrade-1.36-deleting-migrating-workflows-with-preview-profile.adoc[leveloffset=+2]

include::modules/serverless-logic-upgrade-1.36-scaling-down-workflows-with-gitops-profile.adoc[leveloffset=+2]

include::modules/serverless-logic-upgrade-1.36-backing-up-data-index-database.adoc[leveloffset=+2]

include::modules/serverless-logic-upgrade-1.36-backing-up-job-service-database.adoc[leveloffset=+2]

> include::modules/serverless-logic-upgrading-1.36-osl-operator.adoc[leveloffset=+1]

Author: That is correct. I am keeping the leveloffset for +1 because I want to keep the following sections at the same level. I hope that answers your question. :) 

- Preparing for the upgrade
- Upgrading the OpenShift Serverless Logic Operator
- Finalizing the upgrade
